.gitignore

```diff
@@ -3,10 +3,14 @@
 *.war
 *.ear
 
-.idea/workspace.xml
-.idea/libraries
+# Project Files #
+.idea/libraries/
+.idea/shelf/
 .idea/dictionaries/
-.idea/dbnavigator.xml
+.idea/*.xml
+!.idea/codeStyleSettings.xml
+!.idea/encodings.xml
```

Comment:
Do we also want `compiler.xml`, `modules.xml`, `uiDesigner.xml` and/or `vcs.xml`?